### ISSUE TITLE
Access to addDependency

### ISSUE THREAD

**Issue by ryuran** [jabney/render-template-loader#5]
Currently twig files used in `includes` and `extends` are not watched by webpack.
To fix it we need access to `adddependency` https://webpack.js.org/api/loaders/#this-adddependency.

in `twig-html-loader` webpack API is exposed as `context` as parameter of `data` function.

We could add it in as `engine` and/or `locals` function parametter.

**Comment by jabney** [comment]
It sounds like the way to resolve this is to make the [loader context](https://webpack.js.org/api/loaders/#the-loader-context) available to any function that can configure the loader, such as `locals`, `engine`, and `engineOptions`. This way, any of those functions can access `addDependency` and any other method of the loader context.

For backwards compatibility I can do this by setting `this` as the loader context for each of those functions. For example,

```javascript
locals: function () {
  // Access the loader context.
  this.addDependency(...)

  return {
    title: 'Locals as a Function',
    desc: 'The locals option can be an object or function'
  }
}
```

I'll look to address this in `1.2.x`.

**Comment by ryuran** [comment]
Nice!

**Comment by jabney** [comment]
Addressed in `1.2.0`

The loader context is passed to `options` properties which can be functions.

https://github.com/jabney/render-template-loader/blob/0ce8e77b29d33792eea19584d06abd914530f9ab/test/loader.spec.js#L69-L88

Documentation: https://github.com/jabney/render-template-loader#the-loader-context
